### PR TITLE
Refine schedulerQueue test to avoid unneeded locks

### LINKED ISSUE #90754: Refine schedulerQueue unit tests
When reviewing the fix of #88327, I feel that #88331 as well as existing schedulerQueue unit tests are not written in a desired way.

The racing cause of #88327 lies on concurrent access of private fields - those private fields are protected by locks via public functions, but unfortunately directly accessed by the unit tests (under the same package). To dive deeper, it's because the `PriorityQueue` is created and **run immediately**:

https://github.com/kubernetes/kubernetes/blob/de66c643c479f420b1902f8897636ea41ac2b3c3/pkg/scheduler/internal/queue/scheduling_queue_test.go#L1511-L1515

In this way, the unit tests risk us into a situation that two goroutines in `q.Run()` may race with the unit test's calling on private fields/functions.

But, IMO we shouldn't run the flushXYZ() functions **_unconditionally_** for each unit test. Instead, unit tests should be adopting a clear programming model to know what exactly is being tested:

- if it's intended to test the racing with flushXYZ(), we should build dedicated unit tests **without** accessing private fields/functions (or integration tests).
- otherwise, which I think is the common case, don't run the two async flushXYZ() functions; if they have to be called, compose a case by leveraging `FakeClock` and call flushXYZ() explicitly (see [example](https://github.com/kubernetes/kubernetes/blob/de66c643c479f420b1902f8897636ea41ac2b3c3/pkg/scheduler/internal/queue/scheduling_queue_test.go#L1075-L1078)).

/sig scheduling
/cleanup

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/sig scheduling
/kind cleanup

**What this PR does / why we need it**:

- revert PR #88331
- remove createAndRunPriorityQueue()
- PriorityQueue is created and not run by default
- flushXYZ() is called on demand by manipulating FakeClock and Step()

**Which issue(s) this PR fixes**:

Fixes #90754

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/internal/queue/scheduling_queue_test.go [hunk 1]
```diff
@@ -782,6 +765,7 @@ func TestRecentlyTriedPodsGoBack(t *testing.T) {
 		}
 		q.Add(&p)
 	}
+	c.Step(time.Microsecond)
```
Comment (ahg-g) [inline pkg/scheduler/internal/queue/scheduling_queue_test.go:768]: why is this necessary?

File: pkg/scheduler/internal/queue/scheduling_queue_test.go [hunk 2]
```diff
@@ -1021,6 +1006,7 @@ func TestHighPriorityFlushUnschedulableQLeftover(t *testing.T) {
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&highPod), q.SchedulingCycle())
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&midPod), q.SchedulingCycle())
 	c.Step(unschedulableQTimeInterval + time.Second)
```
Comment (ahg-g) [inline pkg/scheduler/internal/queue/scheduling_queue_test.go:1008]: this is not necessary now since flush is called explicitly, right?

**Review by ahg-g — COMMENTED** [review]
couple of minor clarifications.

File: pkg/scheduler/internal/queue/scheduling_queue_test.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:768]: In this case, it's used to make the not-tried Pods' timestamp a bit advanced than the tried Pod (pod-0). Otherwise, even pod-0 is tried once (in `q.Pop()` where its timestamp gets updated), its timestamp holds the same as the other not-tried ones, so in QueueSort, pod-0 would be very likely not popped as expected (we expect it's the last one to be popped).

File: pkg/scheduler/internal/queue/scheduling_queue_test.go (hunk 2 above)
Comment (Huang-Wei) [reply pkg/scheduler/internal/queue/scheduling_queue_test.go:1008]: It's still needed as the flush logic is depending on the time difference:

https://github.com/kubernetes/kubernetes/blob/27f06962d84868be93ead72d0f14eb5922020c4d/pkg/scheduler/internal/queue/scheduling_queue.go#L355-L362

A nit is to update `xyz + time.Second` to `xyz + time.Microsecond` in all places. But it's trivial as they're just fake clocks.
